syntax = "proto3";

package exchangeservice;

service Exchanger {
  rpc Request (ReadRequest) returns (ReadResponse) {}
  rpc Announce (WriteAnnounce) returns (Empty) {}
}

message Empty {}

message ReadRequest {
  string file_name = 1;
  float offset = 2;
  int32 size = 3;
}

message ReadResponse {
  string content = 1;
  int32 size = 2;
}

message WriteAnnounce{
  string worker = 1;
  string file = 2;
}